### PR TITLE
#11: Communication between fragments and changes suggested by sonarlint

### PR DESCRIPTION
**Pull request by viveanban**
*WIP: This PR does not complete the task*
- Communication b/w fragments established through s shared view model and by listening to a live data
- applied some suggestions from sonar lint

some references:
1- https://developer.android.com/training/basics/fragments/communicating#java
2- https://developer.android.com/topic/libraries/architecture/viewmodel#java

### CODE REVIEW

**Comment by thomasgauvin** [comment]
Is this still _WIP_?

File: app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java [hunk 1]
```diff
@@ -315,6 +326,23 @@ public boolean onMarkerClick(Marker marker) {
...
 
+    private void setupPOIListListener() {
+        mViewModel.getListOfPOI().observe(getViewLifecycleOwner(), new Observer<List<WalkingPoint>>() {
+            @Override
+            public void onChanged(List<WalkingPoint> walkingPoints) {
+                for(WalkingPoint point: walkingPoints){
+                    LatLng latLng = new LatLng(point.getCoordinate().getLongitude(),point.getCoordinate().getLatitude());
+                    MarkerOptions markerOptions = new MarkerOptions()
+                            .position(latLng)
+                            //TODO: Change the icon to be the same as the POI icon
```
Comment (Hambrsoom) [inline app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java]: Do you still need this TODO ?

File: app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragmentViewModel.java [hunk 2]
```diff
@@ -218,21 +242,36 @@ public void setFloorPlan(GroundOverlay groundOverlay, String buildingCode, Strin
...
     }
-    public LatLng getSGWZoomLocation(){
+
+    public LatLng getSGWZoomLocation() {
         return buildings.get("H").getCenterCoordinatesLatLng();
     }
 
 
+    public void setListOfPOI(@PoiType String poiType) {
+        //TODO: Filter poi_list to only contain the 10 closest POIs
```
Comment (Hambrsoom) [inline app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragmentViewModel.java]: Do you still need this TODO ?

File: app/src/test/java/com/example/concordia_campus_guide/LocationFragmentViewModelTest.java [hunk 3]
```diff
@@ -25,12 +31,23 @@
...
+    WalkingPointDao mockWalkingPointDao;
+
     private HashMap<String, Building> buildings;
     @Before
     public void init() {
         MockitoAnnotations.initMocks(this);
-        viewModel = new LocationFragmentViewModel();
+        when(mockAppDb.walkingPointDao()).thenReturn(mockWalkingPointDao);
+
+        //TODO: Create test data to test the POI generation
```
Comment (Hambrsoom) [inline app/src/test/java/com/example/concordia_campus_guide/LocationFragmentViewModelTest.java]: Do you still need this TODO?

File: app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java (hunk 1 above)
Comment (viveanban) [reply app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java]: I do. These todos will be cleared before this task is closed. I am keeping them here as checkpoints of what's left to do.

File: app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragmentViewModel.java (hunk 2 above)
Comment (viveanban) [reply app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragmentViewModel.java]: same

File: app/src/test/java/com/example/concordia_campus_guide/LocationFragmentViewModelTest.java (hunk 3 above)
Comment (viveanban) [reply app/src/test/java/com/example/concordia_campus_guide/LocationFragmentViewModelTest.java]: same

File: app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java (hunk 1 above)
Comment (thomasgauvin) [reply app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java]: can you add the issue in the zenhub and add it to the comment please?

Comment (viveanban) [reply app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java]: It's really part of this task tho. Like a subtask. This normally would not be pushed to master but for the sake of smaller more comprehensible PRs I thought of adding it. Do you still think a new issue needs to be created?

Comment (thomasgauvin) [reply app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java]: ok sounds good so I will approve, as long as you push the rest of the code later

**Comment by viveanban** [comment]
Sorry gang! After rebasing on top of master, this branch is crashing and I do not know why. If I don't find the problem, I will need to abandon this PR and I might cherry pick the commits that I am sure are safe.

**Comment by viveanban** [comment]
Turns out I didn't solve the merge conflicts properly when rebasing onto master. Panicked too fast 😅. All good now!

**Pull request merged**